examples/esp_idf/lightdb_stream/main/app_main.c

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2024 Golioth, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "nvs.h"
+#include "shell.h"
+#include "wifi.h"
+#include "freertos/FreeRTOS.h"
+#include "freertos/semphr.h"
+#include <golioth/client.h>
+#include <golioth/stream.h>
+#include <string.h>
+
+#define TAG "lightdb_stream"
+
+struct golioth_client *client;
+static SemaphoreHandle_t _connected_sem = NULL;
+
+static void on_client_event(struct golioth_client *client,
+                            enum golioth_client_event event,
+                            void *arg)
+{
+    bool is_connected = (event == GOLIOTH_CLIENT_EVENT_CONNECTED);
+    if (is_connected)
+    {
+        xSemaphoreGive(_connected_sem);
+    }
+    GLTH_LOGI(TAG, "Golioth client %s", is_connected ? "connected" : "disconnected");
+}
+
+static int get_temperature(float *temp)
+{
+    static int counter = 0;
+
+    /* generate a temperature from 20 deg to 30 deg, with 0.5 deg step */
+
+    *temp = 20.0 + (float)counter / 2;
+
+    counter = (counter + 1) % 20;
+
+    return 0;
```

Comment:
Since this always returns `0`, we can just have this return `temp` instead of an error code and we don't have to pass in the reference.

Reply:
Right